utility/src/main/java/org/oppia/util/parser/UrlImageParser.kt

```diff
@@ -0,0 +1,88 @@
+import org.oppia.util.R
+import javax.inject.Inject
+import javax.inject.Singleton
+
+// TODO(#169): Replace this with exploration asset downloader.
+/** UrlImage Parser for android TextView to load Html Image tag. */
+class UrlImageParser private constructor(
+  @ApplicationContext private val context: Context,
+  @DefaultGcsPrefix private val gcsPrefix: String,
+  @DefaultGcsResource private val gcsResource: String,
+  @ImageDownloadUrlTemplate private var imageDownloadUrlTemplate: String,
+  private val htmlContentTextView: TextView,
+  private val entityType: String,
+  private val entityId: String,
+  @ImageLoaderAnnotation private val imageLoader:ImageLoader
+) : Html.ImageGetter {
+
+
+  /***
+   * This method is called when the HTML parser encounters an <img> tag.
+   * @param urlString : urlString argument is the string from the "src" attribute.
+   * @return Drawable : Drawable representation of the image.
+   */
+  override fun getDrawable(urlString: String): Drawable {
+    imageDownloadUrlTemplate = String.format(imageDownloadUrlTemplate, entityType, entityId, urlString)
+    val urlDrawable = UrlDrawable()
+    val target = BitmapTarget(urlDrawable)
+    imageLoader.load(
+      context,
+      gcsPrefix + gcsResource + imageDownloadUrlTemplate,
+      target
+    )
+    return urlDrawable
+  }
+
+  private inner class BitmapTarget(private val urlDrawable: UrlDrawable) : SimpleTarget<Bitmap>() {
+    internal var drawable: Drawable? = null
+
+    override fun onResourceReady(resource: Bitmap, transition: Transition<in Bitmap>?) {
+      drawable = BitmapDrawable(context.resources, resource)
+      htmlContentTextView.post {
+        val drawableHeight = (drawable as BitmapDrawable).intrinsicHeight
+        val drawableWidth = (drawable as BitmapDrawable).intrinsicWidth
+        val rect = Rect(0, 0, drawableWidth, drawableHeight)
+        (drawable as BitmapDrawable).bounds = rect
+        urlDrawable.bounds = rect
+        urlDrawable.drawable = drawable
+        htmlContentTextView.text = htmlContentTextView.text
+        htmlContentTextView.invalidate()
+      }
+    }
+  }
+
+  private inner class UrlDrawable : BitmapDrawable() {
+    var drawable: Drawable? = null
+    override fun draw(canvas: Canvas) {
+      val currentDrawable = drawable
+      if (currentDrawable != null) {
+        currentDrawable.draw(canvas)
+      }
+    }
+  }
+
+  class Factory @Inject constructor(
+    @ApplicationContext private val context: Context,
+    @DefaultGcsPrefix private val gcsPrefix: String,
+    @DefaultGcsResource private val gcsResource: String,
+    @ImageDownloadUrlTemplate private val imageDownloadUrlTemplate: String,
+    @ImageLoaderAnnotation private val imageLoader: ImageLoader
+  ) {
+    fun create(htmlContentTextView: TextView, entityType: String, entityId: String): UrlImageParser {
+      return UrlImageParser(context,gcsPrefix,gcsResource,imageDownloadUrlTemplate,htmlContentTextView, entityType, entityId,imageLoader)
```

Comment:
Please put spaces between arguments. If you need to line wrap, you can do it like so:

```kotlin
return UrlImageParser(
  context,
  gcsPrefix,
  gcsResource,
  imageDownloadUrlTemplate,
  htmlContentTextView,
  entityType,
  entityId,
  imageLoader
)
```

Reply:
Corrected.